android/src/main/java/com/heapanalytics/reactnative/RNHeapLibraryModule.java

```diff
@@ -81,7 +83,7 @@ private static Map<String, String> convertToStringMap(ReadableMap readableMap) {
         // The JS bridge will flatten maps and arrays in a uniform manner across both
         // platforms.
         // If we get them at this point, we shouldn't continue.
-        throw new RNHeapException("Property objects must be flattened before being sent across the JS bridge.");
+        Log.w("Property objects must be flattened before being sent across the JS bridge. If you get this warning please inspect for non-flattenable objects being sent to Heap");
```

Comment:
Do we have a payload we know causes an error today?  Would it be possible to use that to add an integration test? 

I'd take a look at the integration tests for `getUserId` and `getSessionId`.

The event source could be:

```
      <Button
        title="Send Custom Event"
        onPress={async () =>
          Heap.track('custom-event', { a: {b: null, c: [1,2,3], d: new (class Test {}) } })
        }
      />
```

This produces the event properties `{ 'a.b': null, 'a.c.0': 1, 'a.c.1': 2, 'a.c.2': 3, 'a.d': Test {} }`, which I believe _could_ produce a crash today.  **We should verify if we don't know.**

Integration tests would then validate with:

```
  describe('track', () => {
     it('should serialize properties without crashing', async () => {
       await tapButton('Send Custom Event');

       const message = await tools.server.expectSourceCustomEventWithProperties(
         'custom-event',
         'a.c.0': '1',
         'a.c.1': '2',
         'a.c.1': '3',
       );
     });
   });
```

Reply:
Also, we may want to include the name of the key so developers know where to look.